### ISSUE TITLE
🚀 Feature: Allow Configuration options for catalog-backend-module-msgraph to be modified dynamically

### ISSUE THREAD

**Issue by adawalli** [backstage/backstage#24956]
### 🔖 Feature description

In order to reduce the groups (and transitive users) from large organizations, the `catalog-backend-module-msgraph` provides a nice feature through it's `Config` to specify filters to reduce information with options like [groupFilter](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend-module-msgraph/config.d.ts#L72) and [userGroupMemberFilter](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend-module-msgraph/config.d.ts#L98)

As this is provided as a static config for the deployment, a redeploy is needed when a new group needs to be added. Ideally, the provider config would allow (some) options to be adjusted during the lifetime of a deployment.

### 🎤 Context

As items are ingested into the catalog, it's trivial to add a `processor` that inspects the `spec.owner` field of those entities and identify and `Group` entities that do not yet exist. After verifying that the group is valid within our `entra` tenant, it would be nice to add this group to the filter lists of the running MSGraph Entity Provider.

This is a separate ask issue from the PR addressed in https://github.com/backstage/backstage/pull/23048 as the new groups identified in `spec.owner` may not be nested within another Azure group

### ✌️ Possible Implementation

- Possibly use `extensionPoints` to allow modifications to provider config
- Allow certain properties to be `protected` instead of `private`, which would allow organizations to extend the Graph plugin
- Allow portions of the config to be passed by reference instead of only being pulled from `configRoot`

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!
```[tasklist]
### Tasks
```

**Comment by adawalli** [comment]
This Feature Request is related to a conversation on discord: https://discord.com/channels/687207715902193673/923144214580191282/1245110629686312960

I will state that both @pjungermann and @freben did not like the idea of using `protected` and stated that generally `composition` is preferred, but I don't think that's possible in the plugin as-is

**Comment by Rugvip** [comment]
I think using the subscribe method of the config service could be a good start here? Whenever we're notified that config is changed we can update it as needed.

**Comment by adawalli** [comment]
> I think using the subscribe method of the config service could be a good start here? Whenever we're notified that config is changed we can update it as needed.

That sounds interesting - is that something that exists today?

**Comment by adawalli** [comment]
I looked around a bit, and although I might have missed it, I couldn’t see this being used in any plugins today. The only real use-case I found was for cases where sensitive secrets might be hidden (which is fine in a “one-time” type fashion) or in another case possibly for something like a DB connection where you define a JS token receiver through the app_config file. Are there some other examples I might be able to look at?
